### PR TITLE
Issue #15040: Add support for yield in SimplifyBooleanReturnCheck

### LINKED ISSUE #15040: Add Check Support for Java 21 Pattern Matching for Switch Syntax: SimplifyBooleanReturn
child of https://github.com/checkstyle/checkstyle/issues/14961

I have read check documentation:https://checkstyle.org/checks/coding/simplifybooleanreturn.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test.java                                                
record ColoredPoint(boolean p, int x, String c) { }
record Rectangle(boolean upperLeft, boolean lowerRight) { }
public class Test {

    boolean test1(Object o) {
       if (o instanceof ColoredPoint(boolean x, _, _)) {
            if (x) return true;   // violation
            else return false;
        } else {
            return false;
        }
    }
    boolean test2(Object obj) {
        return switch (obj) {
            case ColoredPoint(boolean x,_,_) -> {
                if (x) yield true;       // expected violation , can be yield x
                else yield false;
            }
            default -> false;
        };
    }
}
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="SimplifyBooleanReturn">

        </module>
    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:7:9: Conditional logic can be removed. [SimplifyBooleanReturn]
Audit done.
Checkstyle ends with 1 errors.
PS D:\CS\test> 
```

Describe what you want in detail

I am expecting a violation on unnecessary conditional logic that can be simplified inside switch expressions. each case block is an execution path for a return. So this condition falls under a Boolean return

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #15040

### CODE REVIEW

**Comment by mahfouz72** [comment]
Github, generate report for SimplifyBooleanReturn/Example1

**Comment by mahfouz72** [comment]
github, generate site

**Comment by mahfouz72** [comment]
Github, generate report for SimplifyBooleanReturn/Example1

**Comment by romani** [comment]
In report there is no diff, mostly because code bases that we use don't use modern syntax.

Please search in GitHub java project where yield is in use.

**Comment by mahfouz72** [comment]
Another reason is that we are looking at a very specific code which is :

```Java
if (xxx) {
  yield true;
}
else {
 yield false;
}
```

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/InputSimplifyBooleanReturnWithYield.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+SimplifyBooleanReturn
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.simplifybooleanreturn;
+
+public class InputSimplifyBooleanReturnWithYield {
```
Comment (nrmancuso) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/InputSimplifyBooleanReturnWithYield.java:9]: Please add inputs with case labels (`:`) and also equivalent switches (where possible) with `return`.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
It is unlikely that we will find much usage like this in project files.

Item:

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/InputSimplifyBooleanReturnWithYield.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/InputSimplifyBooleanReturnWithYield.java:9]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 2]
```diff
@@ -140,20 +140,21 @@ private static boolean canReturnOnlyBooleanLiteral(DetailAST ast) {
...
      * <pre>
-     * return true/false;
+     * return/yield true/false;
      * </pre>
      *
      * @param ast the syntax tree to check
-     * @return if ast is a return statement with a boolean literal.
+     * @return if ast is a return or a yield statement with a boolean literal.
      */
     private static boolean isBooleanLiteralReturnStatement(DetailAST ast) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java:153]: I find it odd you updated the javadocs to make it clear it is a return or yield, but then you leave the method as `ReturnStatement`.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java (hunk 2 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: done

**Review by rnveach — APPROVED** [review]

**Pull request merged**